LLama/LLamaInteractExecutor.cs

```diff
@@ -154,15 +155,21 @@ private Task PreprocessLlava(string text, InferStateArgs args, bool addBos = tru
                 {
                     if (image.Type == ImageData.DataType.ImagePath && image.Data != null)
```

Comment:
Can all of this logic be moved out of the executor? Maybe a base interface with separate classes for the different variations.

For example:

e.g.
```csharp
interface IClipImage { Task<SafeLlavaImageEmbedHandle> GetEmbed(); }

class ImageDataFromUrl(string url)
{
    public async Task<SafeLlavaImageEmbedHandle> GetEmbed()
    {
        return SafeLlavaImageEmbedHandle.CreateFromMemory(await GetImageBytes());
    }

    private async Task<byte[]> GetImageBytes()
    {
        return await DownloadThatUrl(url);
    }
}
```

Reply:
I was thinking the same after adding more and more to the code. The reading/downloading of the image should be the task of the user of the library.